lib/responses.js

```diff
@@ -328,6 +351,7 @@ function clientError(err, res) {
 function handleError(res) {
   return function (err) {
     if (err.status && err.name !== 'NotFoundError') {
+      log('error', err.message, { code: err.status, stack: err.stack });
```

Comment:
I'm hesitant about adding logging here because this `if` statement specifically deals with errors that have already been handled to some extent.

Assigning a `status` to an `error` is a way to explicitly tell Amphora how to communicate the error in the response. The only way to get here is by:

1. Manually constructing and throwing an `Error`.
2. `try/catch` an `Error` and setting `.status`.

We could risk duplicate logs or logging handled events at the error level by putting this line here.

Reply:
@mattoberle I tracked down all the places where this function was called. None of them log errors before this function is called.

Reply:
If I'm looking at the right places, it looks like the few spot this is called in `amphora` do not set a `.status`, meaning that they hit the `clientError` or `serverError` functions (which do log).

The block with the new `log` call is intended to be a way to force a particular response code by throwing somewhere where direct access to `res` is unavailable.

eg. Maybe in a `model.save` method one would do:

```js
if (validationFailed) {
  const err = new Error('Invalid value for X');
  err.status = 400;
  throw err;
}
```

If you find a particular location in this repository where we are setting `.status` I'm 100% open to considering logging where the error is constructed-- I just want to avoid the duplication consequences for anyone taking advantage of this as-implemented!

Reply:
Where would the duplication be? If anyone is setting err.status, they would also have to be manually using clay log to log the error in order for it to be duplicated right?

Reply:
You are right, the duplication would be manual-- but opting in to log at the time of error handling gives the implementer a choice of:

1. What context to include.
2. What level to log at. 

If I log when I'm handling the error I can do something like this:

```js
log('debug', 'Invalid value for X', { x });
```

Doing that groups all like-messages under `.msg == "Invalid value for X"` and supplies the context of what `x` was. It also let's me use the `debug` log-level, because it's not something we usually need to know about.

With the logging built-in to amphora I can't choose the log level and the only way to provide context is to set the message like this:
```js
Error(`Invalid value for X: ${x}`);
```